internal/venice-common/src/main/java/com/linkedin/venice/system/store/MetaStoreWriter.java

```diff
@@ -494,15 +500,31 @@ private void closeVeniceWriter(String metaStoreName, VeniceWriter veniceWriter,
 
   @Override
   public void close() throws IOException {
-    // Close VeniceWrites in parallel to reduce the time to shut down the server.
-    try (Timer ignore = Timer.run(
-        elapsedTimeInMs -> LOGGER.info(
-            "MetaStoreWriter takes {}ms to close {} VeniceWriters in parallel",
-            elapsedTimeInMs,
-            metaStoreWriterMap.size()))) {
-      metaStoreWriterMap.entrySet()
-          .parallelStream()
-          .forEach(entry -> closeVeniceWriter(entry.getKey(), entry.getValue(), false));
+    long startTime = System.currentTimeMillis();
+    LOGGER.info("Closing MetaStoreWriter - numOfVeniceWriters: {}", metaStoreWriterMap.size());
+    // iterate through the map and close all the VeniceWriters
+    List<CompletableFuture<VeniceResourceCloseResult>> closeFutures = new ArrayList<>(metaStoreWriterMap.size());
+
+    for (Map.Entry<String, VeniceWriter> entry: metaStoreWriterMap.entrySet()) {
+      closeFutures.add(entry.getValue().closeAsync(true));
```

Comment:
One drawback of this is that it will create one thread per `VW::closeAsync` and in some cases it could overwhelm system for a short period of time. Maybe we should update this to have more control over number of inflight close operations.

Reply:
Doesn't it actually need 2 threads per `closeAsync` (or per VeniceWriter)? Yeah, it sounds like a good idea to bound the max number of threads for closing writers.

Reply:
done